sdk/search/azure-search/src/main/java/com/azure/search/SearchServiceAsyncClient.java

```diff
@@ -0,0 +1,1759 @@
+     *
+     * @param indexName the name of the index for which to retrieve statistics
+     * @return the index statistics result.
+     */
+    public Mono<GetIndexStatisticsResult> getIndexStatistics(String indexName) {
+        try {
+            return this.getIndexStatisticsWithResponse(indexName, null)
+                .map(Response::getValue);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Returns statistics for the given index, including a document count and storage usage.
+     *
+     * @param indexName the name of the index for which to retrieve statistics
+     * @param requestOptions additional parameters for the operation.
+     *                       Contains the tracking ID sent with the request to help with debugging
+     * @return a response containing the index statistics result.
+     */
+    public Mono<Response<GetIndexStatisticsResult>> getIndexStatisticsWithResponse(String indexName,
+                                                                                   RequestOptions requestOptions) {
+        try {
+            return withContext(context -> this.getIndexStatisticsWithResponse(indexName, requestOptions, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<GetIndexStatisticsResult>> getIndexStatisticsWithResponse(String indexName,
+                                                                            RequestOptions requestOptions,
+                                                                            Context context) {
+        return restClient
+            .indexes()
+            .getStatisticsWithRestResponseAsync(indexName, requestOptions, context)
+            .map(Function.identity());
+    }
+
+    /**
+     * Lists all indexes available for an Azure Cognitive Search service.
+     *
+     * @return a reactive response emitting the list of indexes.
+     */
+    public PagedFlux<Index> listIndexes() {
+        try {
+            return this.listIndexes(null, null);
+        } catch (RuntimeException ex) {
+            return new PagedFlux<>(() -> monoError(logger, ex));
+        }
+    }
+
+    /**
+     * Lists all indexes available for an Azure Cognitive Search service.
+     *
+     * @param select selects which top-level properties of the index definitions to retrieve.
+     *               Specified as a comma-separated list of JSON property names, or '*' for all properties.
+     *               The default is all properties
+     * @param requestOptions additional parameters for the operation.
+     *                       Contains the tracking ID sent with the request to help with debugging
+     * @return a reactive response emitting the list of indexes.
+     */
+    public PagedFlux<Index> listIndexes(String select, RequestOptions requestOptions) {
```

Comment:
@navalev Why do the synchronous methods have a `Context` parameter, but the asynchronous methods don't?

Reply:
Following the guidlines https://azure.github.io/azure-sdk/java_design.html#java-service-client-context:
DO provide an overload method that takes a com.azure.core.util.Context argument for each service operation in sync clients only